resources/build/zip.inc.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+#
+# This script contains zip and unzip utilities to function for all environments.
+# unzip is available in all environments
+# zip is not available by default on Windows, but can be manually installed.
+# If zip is not available, then fall back to 7z for zipping.
+#
+
+# convert parameters into array
+# $1 - count of parameters
+# $@ - remaining parameters to convert into array
+# returns array of strings
+function _parse_args() {
+  count=$1
+  shift
+  value=()
+  for (( i=0; i<$count; i++)); do
+    value+=($1)
+    shift
+  done
+  echo ${value[@]}
+}
+
+# [--name zip filename]
+# [--flags   n  list of n flags to pass to zip command]
+# [--include n  list of n files to include in zip]
+# [--exclude n  list of n files to exclude from zip]
```

Comment:
For the caller it might be easier instead of passing the number of files to include as first parameter and then the files, to treat all following args as files to include until we encounter an arg that starts with `--` or we're out of args.

That would be easier to maintain in the calling script because you don't have to remember to update the number, and it would be easier in case we use wild cards that the shell expands.

Reply:
I'd still go with the previous suggestion.